File: src/main/java/com/bridgecrew/util/ApplicationServiceUtil.java
package com.bridgecrew.util;

import com.intellij.openapi.application.ApplicationManager;
import lombok.experimental.UtilityClass;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@UtilityClass
public class ApplicationServiceUtil {

    private static final Logger logger = LoggerFactory.getLogger(ApplicationServiceUtil.class);

    public static <T> T getService(Class<T> clazz) {
        T service = ApplicationManager.getApplication().getService(clazz);
        if (service == null) {
            String message = String.format("Could not find service %s", clazz.getSimpleName());
            logger.error(message);
            throw new IllegalArgumentException(message);
        }
        return service;
    }
}

File: src/main/kotlin/com/bridgecrew/utils/ApplicationServiceUtil.kt
package com.bridgecrew.utils

import com.intellij.openapi.application.ApplicationManager
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class ApplicationServiceUtil {

    companion object {

        val logger: Logger = LoggerFactory.getLogger(ApplicationServiceUtil::class.java)

        @JvmStatic
        fun <T> getService(clazz: Class<T>): T {
            val service = ApplicationManager.getApplication().getService(clazz)
            if (service == null) {
                val message = String.format("Could not find service %s", clazz.simpleName)
                logger.error(message)
                throw IllegalArgumentException(message)
            }
            return service
        }
    }
}